test/integration/endpointslice/endpointsliceterminating_test.go

```diff
@@ -0,0 +1,333 @@
+					Name: "fake-node",
+				},
+			}
+
+			_, err = client.CoreV1().Nodes().Create(context.TODO(), node, metav1.CreateOptions{})
+			if err != nil {
+				t.Fatalf("Failed to create test node: %v", err)
+			}
+
+			svc := &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-service",
+					Namespace: ns.Name,
+					Labels: map[string]string{
+						"foo": "bar",
+					},
+				},
+				Spec: corev1.ServiceSpec{
+					Selector: map[string]string{
+						"foo": "bar",
+					},
+					Ports: []corev1.ServicePort{
+						{Name: "port-443", Port: 443, Protocol: "TCP", TargetPort: intstr.FromInt(443)},
+					},
+				},
+			}
+
+			_, err = client.CoreV1().Services(ns.Name).Create(context.TODO(), svc, metav1.CreateOptions{})
+			if err != nil {
+				t.Fatalf("Failed to create test Service: %v", err)
+			}
+
+			pod := &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-pod",
+					Labels: map[string]string{
+						"foo": "bar",
+					},
+				},
+				Spec: corev1.PodSpec{
+					NodeName: "fake-node",
+					Containers: []corev1.Container{
+						{
+							Name:  "fakename",
+							Image: "fakeimage",
+							Ports: []corev1.ContainerPort{
+								{
+									Name:          "port-443",
+									ContainerPort: 443,
+								},
+							},
+						},
+					},
+				},
+			}
+
+			pod, err = client.CoreV1().Pods(ns.Name).Create(context.TODO(), pod, metav1.CreateOptions{})
+			if err != nil {
+				t.Fatalf("Failed to create test ready pod: %v", err)
+			}
+
+			pod.Status = testcase.podStatus
+			_, err = client.CoreV1().Pods(ns.Name).UpdateStatus(context.TODO(), pod, metav1.UpdateOptions{})
+			if err != nil {
+				t.Fatalf("Failed to update status for test ready pod: %v", err)
+			}
+
+			// first check that endpoints are included, test should always have 1 initial endpoint
+			err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+				esList, err := client.DiscoveryV1().EndpointSlices(ns.Name).List(context.TODO(), metav1.ListOptions{
+					LabelSelector: discovery.LabelServiceName + "=" + svc.Name,
+				})
+
+				if err != nil {
+					return false, err
+				}
+
+				if len(esList.Items) == 0 {
+					return false, nil
+				}
+
+				numEndpoints := 0
+				for _, slice := range esList.Items {
+					numEndpoints += len(slice.Endpoints)
+				}
+
+				if numEndpoints > 0 {
+					return true, nil
+				}
+
+				return false, nil
+			})
+			if err != nil {
+				t.Errorf("Error waiting for endpoint slices: %v", err)
+			}
+
+			// Delete pod and check endpoints slice conditions
+			err = client.CoreV1().Pods(ns.Name).Delete(context.TODO(), pod.Name, metav1.DeleteOptions{})
+			if err != nil {
+				t.Fatalf("Failed to delete pod in terminating state: %v", err)
+			}
+
+			// Validate that terminating the endpoint will result in the expected endpoints in EndpointSlice.
+			// Use a stricter timeout value here since we should try to catch regressions in the time it takes to remove terminated endpoints.
```

Comment:
Is 10 seconds strict here? Do we think there's any chance this will result in flakes?